cdap-ui/app/cdap/components/DataPrep/store/index.js

```diff
@@ -100,7 +101,23 @@ const error = (state = errorInitialState, action = defaultAction) => {
     case DataPrepActions.setError:
       stateCopy = Object.assign({}, state, {
         showError: true,
-        message: action.payload.message
+        message: action.payload.message,
+        cliError: null
+      });
+      break;
+    case DataPrepActions.setCLIError:
+      stateCopy = Object.assign({}, state, {
+        showError: false,
+        message: '',
+        cliError: action.payload.message
+      });
+      break;
+    case DataPrepActions.setWorkspace:
+    case DataPrepActions.setDirectives:
+      stateCopy = Object.assign({}, state, {
+        showError: false,
+        message: '',
+        cliError: null
```

Comment:
Can we use the `showError` state similar to how you are using the `cliError` state right now, and remove the `message` state? e.g. `showError: null`. Seems like this will reduce redundant code.